pkg/bytecode/vm.go

```diff
@@ -12,8 +13,14 @@ const (
 	GlobalsSize = 65536
 )
 
-// ErrStackOverflow is returned when the stack exceeds its size limit.
-var ErrStackOverflow = errors.New("stack overflow")
+var (
+	// ErrStackOverflow is returned when the stack exceeds its size limit.
+	ErrStackOverflow = errors.New("stack overflow")
```

Comment:
IMO this should wrap `ErrPanic`, `ErrRuntime` or similar.

In the evaluator I called it [`ErrPanic`](https://github.com/evylang/evy/blob/main/pkg/evaluator/evaluator.go#L19), things that can go wrong during runtime
such as `ErrBounds` for index out of bounds.

Then you have 2 top level error types:

- `ErrInternal` - should never happen, we mucked something up.
- `ErrPanic` or similar - "user error".

Reply:
I'm looking at the evaluator again and I'm wondering why I called this sentinel
`ErrPanic`, especially as I also have a `PanicError` that is returned by Evy's
builtin `panic "boom"` function.

The problem with the name `ErrRuntime` is that while these errors happen at "execution time" and are not preventable (ie not internal), they don't align with what Runtime means in a different Evy execution context: 
In this codebase the Runtime means "environment providing hooks" -  browser or the terminal - one provides graphics functions and means to write to an HTML `<div>` the other one is purely text based using Stdin/Stdout. 

So if you have a good idea wrt to top level error grouping / naming - go ahead and I'll adjust inside the evaluator.